src/Workspaces/Core/Portable/Options/OptionServiceFactory.cs

```diff
@@ -40,11 +40,6 @@ internal class OptionService : IWorkspaceOptionService
             private readonly IGlobalOptionService _globalOptionService;
             private readonly IWorkspaceTaskScheduler _taskQueue;
 
-            /// <summary>
-            /// Gate guarding <see cref="_eventHandlers"/> and <see cref="_documentOptionsProviders"/>.
-            /// </summary>
-            private readonly object _gate = new object();
```

Comment:
is this collection being mutated so much that this gate was a problem?  I'm curious where the contention is coming from.